lib/tasks/demo-build.js

```diff
@@ -88,29 +89,62 @@ function buildJs(buildConfig) {
 	}
 }
 
+function loadRemoteDemoData(url) {
+	const genericErrorMessage = 'Could not load remote demo data.';
+	return request.get({
+		uri: url,
+		json: true
+	})
+	.then(data => {
+		if (typeof data === 'object') {
+			return data;
+		}
+		throw new Error(`${genericErrorMessage} ${url} did not provide valid JSON.`);
+	}).catch((err) => {
+		if (err.name === 'RequestError') {
+			err.message = `${genericErrorMessage} ${url} does not appear to be valid.`;
+		}
+		if (err.name === 'StatusCodeError') {
+			  err.message = `${genericErrorMessage} ${url} returned a ${err.statusCode} status code.`;
+		}
+		const e = new Error(err.message);
+		e.stack = '';
```

Comment:
Why do we remove the stack from the error here? Could be useful in debugging

Reply:
Good question. Me and @JakeChampion chatted about that for a while. I think we opted for a more concise error to make OBT more friendly `Could not load remote demo data. ${remoteDataUrl} returned a ${statusCode} status code.` is straight to the point, a stack trace of vendor code is not so helpful. Perhaps the best-of-both would be to remove the stack trace only for potentially common cases we handle `RequestError` `StatusCodeError`.

Reply:
But with a new `Error` thrown (the one you create) it doesn't include any vendor code in the stack trace anyway right?

Just feels like this library code maybe isn't the place to decide whether the stack trace is useful to a dev who's trying to debug later 🙂. If we're stripping stack traces from certain errors to make the output more friendly, should that be handled by the [command line part of OBT](https://github.com/Financial-Times/origami-build-tools/blob/major/lib/origami-build-tools-cli.js#L115)?